returnn/rasr_training.py

```diff
@@ -232,9 +234,22 @@ def create_config(
         return config, post_config
 
     @classmethod
-    def create_dataset_config(cls, train_crp, partition_epochs):
+    def create_dataset_config(cls, train_crp, returnn_config, partition_epochs):
+        """
+        :param rasr.CommonRasrParameters train_crp:
+        :param ReturnnConfig returnn_config:
+        :param dict[str, int] partition_epochs:
+        :return:
+        """
         datasets = {}
 
+        assert not (
+            "partition_epoch" in returnn_config.config["train"] and partition_epochs
```

Comment:
What about making `partition_epochs` an empty dict by default and then check
```
assert not (
            "partition_epoch" in returnn_config.config["train"] and "train" in partition_epochs
)
```
and the same for dev?

I know, it would be strange to define partition_epoch partially in one variable and partially in another variable, but at least you remove the bad None default.

Reply:
Why is the `None` default worse than an empty dict?

Reply:
Because every time we don't let something be None is a potential source of errors removed.

Reply:
It is not good style to have mutable objects as default parameters, since the object will be created once and reused across all calls. If for example the constructor were to change the dict it would also change for all future evaluations. This is why we usually use None as a default parameter.

Reply:
Ok sorry, I didn't know that default parameters are shared between function calls.